src/observable/fromArrayLike.ts

```diff
@@ -0,0 +1,66 @@
+import {Scheduler} from '../Scheduler';
+import {Observable} from '../Observable';
+import {Subscriber} from '../Subscriber';
+import {Subscription} from '../Subscription';
+
+export class ArrayLikeObservable<T> extends Observable<T> {
+
+  private mapFn: (x: number, y: any) => T;
+
+  static create<T>(arrayLike: ArrayLike<T>, mapFn: (x: number, y: any) => T, thisArg: any, scheduler?: Scheduler) {
+    return new ArrayLikeObservable(arrayLike, mapFn, thisArg, scheduler);
+  }
+
+  static dispatch(state: any) {
+
+    const { arrayLike, index, count, mapFn, subscriber } = state;
+
+    if (index >= count) {
+      subscriber.complete();
+      return;
+    }
+
+    const result = mapFn ? mapFn(index, arrayLike[index]) : arrayLike[index];
+    subscriber.next(result);
+
+    if (subscriber.isUnsubscribed) {
```

Comment:
I think we want to check `isUnsubscribed` first thing in here. Because anything could have happened to unsubscribe prior to `dispatch` being called. It won't hurt much, as-is, because we don't allow `next` to do anything if it's already unsubscribed, but it would be more efficient to check at the top of this function.